Task_2/src/Pages/root.tsx

```diff
@@ -1,12 +1,88 @@
 import { Outlet } from "react-router-dom";
 import React from "react";
 import Navbar from "../components/Navbar/Navbar";
+import CartContext from "../Store/ShoppingCart";
+import AuthenticationContext from "../Store/Authentication";
+import { useState } from "react";
+
 const RootLayout: React.FC = () => {
+  const [cart, setCart] = useState<{ items: { id: number; count: number }[] }>({
+    items: [],
+  });
+
+  const [authData, setAuthData] = useState<{
+    isAuthenticated: string | null;
+    token: string | null;
+  }>({
+    isAuthenticated: localStorage.getItem("isAuthenticated"),
+    token: localStorage.getItem("token"),
+  });
+
+  const addItemToCartHandler = (id: number) => {
+    const { items } = cart;
+    for (let i = 0; i < items.length; i++) {
+      if (items[i].id === id) {
+        return;
+      }
+    }
+    items.push({ id: id, count: 1 });
+    setCart({ items: items });
+  };
+
+  const deleteItemFromCart = (id: number) => {
+    const { items } = cart;
+    for (let i = 0; i < items.length; i++) {
```

Comment:
@codal-uchoudhary same here it can be optimized with simply using **filter** method.

Reply:
done